### PR TITLE
add kinesis stream scaler content

### PR DESCRIPTION
**Pull request by msfuko**
The doc for kedacore/keda#526.

### CODE REVIEW

File: content/scalers/aws-kinesis.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
++++
+fragment = "content"
+weight = 100
+title = "AWS Kinesis Stream"
+background = "light"
++++
+
+Scale applications based on AWS Kinesis Stream.
+
+**Availability:** v1.0+ | **Maintainer:** Community
```
Comment (tomkerkhove) [inline content/scalers/aws-kinesis.md]: This looks like a copy/paste mistake, can you change it to 1.1+ please?
```suggestion
**Availability:** v1.1+ | **Maintainer:** Community
```

File: content/scalers/aws-kinesis.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+<!--more-->
+
+### Trigger Specification
+
+This specification describes the `aws-kinesis-stream` trigger that scales based on the shard count of AWS Kinesis Stream.
+
+```yaml
+triggers:
+  - type: aws-sqs-queue
```
Comment (tomkerkhove) [inline content/scalers/aws-kinesis.md]: This doesn't look like the right scaler type
```suggestion
  - type: aws-kinesis-stream
```

File: content/scalers/aws-kinesis.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    metadata:
+      # Required: streamName
+      streamName: myKinesisStream
+      # Required: awsRegion
```
Comment (tomkerkhove) [inline content/scalers/aws-kinesis.md]: Field below has same name, no need to duplicate
```suggestion
      # Required
```

File: content/scalers/aws-kinesis.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+### Trigger Specification
+
+This specification describes the `aws-kinesis-stream` trigger that scales based on the shard count of AWS Kinesis Stream.
+
+```yaml
+triggers:
+  - type: aws-sqs-queue
+    metadata:
+      # Required: streamName
```
Comment (tomkerkhove) [inline content/scalers/aws-kinesis.md]: Field below has same name, no need to duplicate
```suggestion
      # Required:
```

File: content/scalers/aws-kinesis.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      streamName: myKinesisStream
+      # Required: awsRegion
+      awsRegion: "eu-west-1"
+      # Optional: Default: 2
+      shardCount: "2"
+```
+
+**Parameter list:**
+
+- `streamName` - Kinesis Stream name
```
Comment (tomkerkhove) [inline content/scalers/aws-kinesis.md]: Nipicking suggestion to improve readability:
```suggestion
- `streamName` - Name of AWS Kinesis Stream
```

File: content/scalers/aws-kinesis.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+**Role based authentication:**
+
+- `awsRoleArn` - Amazon Resource Names (ARNs) uniquely identify AWS resource
+
+**Credential based authentication:**
+
+- `awsAccessKeyID` - Id of the user
+- `awsSecretAccessKey` - Access key for the user to authenticate with
+
+The user will need access to read data from AWS Cloudwatch.
```
Comment (tomkerkhove) [inline content/scalers/aws-kinesis.md]: Is this still applicable to get the metrics via Cloudwatch or is this a copy/paste mistake?

File: content/scalers/aws-kinesis.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+**Parameter list:**
+
+- `streamName` - Kinesis Stream name
+- `shardCount` - The target value that a Kinesis data streams consumer can handle.
+- `awsRegion` - AWS Region for the Kinesis Stream
+
+### Authentication Parameters
+
+You can use `TriggerAuthentication` CRD to configure the authenticate by providing either a role ARN or a set of IAM credentials.
```
Comment (tomkerkhove) [inline content/scalers/aws-kinesis.md]: TriggerAuth CRD is optional so we might want to clarify this here.

See https://keda.sh/concepts/authentication/

Comment (msfuko) [reply content/scalers/aws-kinesis.md]: I updated the description and added the link https://github.com/kedacore/keda-docs/pull/61/files#diff-664a100428eb94713fde5e3721428185R38
anything could be improved? thanks!

**Comment by msfuko** [comment]
Hey @tomkerkhove Thanks for the code review! I've updated accordingly.
The change isn't huge so I merge them into a single commit. Please let me know if I should just submit the separate commit.
Thanks again for your time!

File: content/scalers/aws-kinesis.md (hunk 7 above)
Comment (tomkerkhove) [reply content/scalers/aws-kinesis.md]: Thanks!

File: content/scalers/aws-kinesis.md (hunk 6 above)
Comment (tomkerkhove) [reply content/scalers/aws-kinesis.md]: Let me know what you think here @msfuko and should be good to go

Comment (msfuko) [reply content/scalers/aws-kinesis.md]: Originally I removed the whole sentence. to be more precise, I added "The user will need `DescribeStreamSummary` access to read data from AWS Kinesis Streams." here. hope this looks good.

Comment (tomkerkhove) [reply content/scalers/aws-kinesis.md]: Thanks!

**Pull request merged**